src/dataExplorer/context/persistance.tsx

```diff
@@ -126,23 +150,23 @@ export const PersistanceProvider: FC = ({children}) => {
     [hasChanged]
   )
 
-  const clearSchemaSelection = () => {
-    setSelection(JSON.parse(JSON.stringify(DEFAULT_SCHEMA)))
+  const clearCompositionSelection = () => {
+    setSelection(JSON.parse(JSON.stringify(DEFAULT_SELECTION)))
   }
 
-  const setSchemaSelection = useCallback(
-    schema => {
-      if (selection.composition?.diverged && schema.composition?.synced) {
+  const setCompositionSelection = useCallback(
+    newSelection => {
+      if (selection.composition?.diverged && newSelection.composition?.synced) {
         // cannot re-sync if diverged
         return
       }
-      const nextState: SchemaSelection = {
+      const nextState: CompositionSelection = {
         ...selection,
-        ...schema,
+        ...newSelection,
         composition: {
           ...(selection.composition || {}),
-          ...(schema.composition || {}),
-        },
+          ...(newSelection.composition || {}),
```

Comment:
`newSelection.composition` is a nested object, which is why it's a spread update. Since the new `resultOptions` are also nested, then those would also need to be a spread update. Additionally, any updates to the `resultOptions` (or`resultOptions.group`) should replace the entire resultOptions pointer.

a.k.a. the same way that the `newSelection.composition` works.

In case anyone wants history on why we are using `RecursivePartial<CompositionSelection>` types in this callback, the reason is to avoid memoization resetting old values. As per this ticket:
https://github.com/influxdata/ui/issues/5708